### ISSUE TITLE
LogAndConfigCmd and _cat/nodes config location incompatibility

### ISSUE THREAD

**Issue by tsouza** [elastic/elasticsearch-support-diagnostics#152]
Starting with ES 5.3 (at least) _cat/nodes provides a different structure for directory paths.

The incompatibility makes the diagnostics tool throw an error:
```
java.io.FileNotFoundException: Source '/usr/share/elasticsearch/config' does not exist
        at org.apache.commons.io.FileUtils.copyDirectory(FileUtils.java:1368) ~[commons-io-2.4.jar:2.4]
        at com.elastic.support.diagnostics.commands.LogAndConfigCmd.execute(LogAndConfigCmd.java:85) ~[support-diagnostics-5.11.jar:5.11]
        at com.elastic.support.diagnostics.commands.AbstractDiagnosticCmd.execute(AbstractDiagnosticCmd.java:15) [support-diagnostics-5.11.jar:5.11]
        at com.elastic.support.diagnostics.chain.Chain.execute(Chain.java:33) [support-diagnostics-5.11.jar:5.11]
        at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:41) [support-diagnostics-5.11.jar:5.11]
```

According to [LogAndConfigCmd.java#L144](https://github.com/elastic/elasticsearch-support-diagnostics/blob/master/src/main/java/com/elastic/support/diagnostics/commands/LogAndConfigCmd.java#L144). It expects something like:
```json
"path" : {
   "logs" : "/var/log/elasticsearch",
    "home" : "/usr/share/elasticsearch",
    "conf": "/etc/elasticsearch"
},
```

But now it seems actually to be:
```json
       "path" : {
          "logs" : "/var/log/elasticsearch",
          "home" : "/usr/share/elasticsearch"
        },
        "default" : {
          "path" : {
            "data" : "/var/lib/elasticsearch",
            "logs" : "/var/log/elasticsearch",
            "conf" : "/etc/elasticsearch"
          }
        }
```

**Comment by nellicus** [comment]
https://github.com/elastic/elasticsearch-support-diagnostics/blob/master/src/main/java/com/elastic/support/diagnostics/commands/LogAndConfigCmd.java#L144 expects three strings as a function, I am not sure I understand what you mean with 


>it expects something like:

```
"path" : {
   "logs" : "/var/log/elasticsearch",
    "home" : "/usr/share/elasticsearch",
    "conf": "/etc/elasticsearch"
},
```

btw if you run the diag remotely you need to pass --type remote 
otherwise it's working as designed and you get
```
java.io.FileNotFoundException: Source '/usr/share/elasticsearch/config' does not exist
```

**Comment by tsouza** [comment]
@nellicus The "conf" parameter of the method comes from "path.conf" (https://github.com/elastic/elasticsearch-support-diagnostics/blob/master/src/main/java/com/elastic/support/diagnostics/commands/LogAndConfigCmd.java#L66). So there ain't no `path.conf` in this case (because it is actually in `default.path.conf`) and then the implementation will return `path.home` + `/config`.

**Comment by nemonster** [comment]
The issue is that for defaults, if there is no explicit setting for logs the default value is copied to path.logs. But this doesn't happen for path.conf. Which I happen to think is inconsistent, but I've put defensive logic into the diag to check for the path.conf and path.logs first, and if they don't exist try the default.path.x trees before constructing one.

Resolved in the 5.12 release.
